src/Illuminate/Foundation/Testing/TestResponse.php

```diff
@@ -685,17 +703,22 @@ public function assertJsonMissingValidationErrors($keys)
      */
     public function decodeResponseJson($key = null)
     {
-        $decodedResponse = json_decode($this->getContent(), true);
+        $decodedResponseStdClass = (array) json_decode($this->getContent());
+        $decodedResponseAssoc = json_decode($this->getContent(), true);
```

Comment:
This should be moved just before `$decodedResponse =` as we might fail/throw an exception.

Reply:
```
$decodedResponseAssoc = json_decode($this->getContent(), true);
$decodedResponseStdClass = (array) json_decode($this->getContent());
```
such ?

Reply:
`$decodedResponseAssoc` is useless in this context and should be moved down